### ISSUE TITLE
[QUESTION]: High SD using k-fold CV. How to deal with it?

### ISSUE THREAD

**Issue by spyda90** [chemprop/chemprop#592]
Hi,

I am attempting to build a predictive model of pIC50 against a certain drug target of interest. I have an in-hose dataset of about 400 datapoints.
Many of them are inactives, so I tried to use the bounded_mse and bounded_rmse loss of function and metric, respectively, indicating inactive compounds as <4.5.

During hyperparameter optimization and training I changed some settings (# iterations, Rdkit features Y/N), keeping always ensemble (5) and cross-validation (k-fold 10). Splitting data 80:10:10 (--split-type cv).

The results of the internal validation set (the last 10%) are good in term of RMSE (<0.45) but they have high SD (around 0.45). This happens because the different folds gave very different results, ranging from 0.2 to 1.5 RMSE value.

In my opinion this might happen because in some fold there were very few active cpds. 

In your opinion, reducing the k value of cross validation (or avoiding it) may be sufficient to reduce SD? Otherwise, how can I try?

Thank you in advance.
Francesco

**Comment by JacksonBurns** [comment]
> In my opinion this might happen because in some fold there were very few active cpds.

This is probably the case. At the beginning of a run, chemprop should print some basic info about the dataset, including the frequency of labels during classification. Does your dataset have a small % of active compounds here?

> In your opinion, reducing the k value of cross validation (or avoiding it) may be sufficient to reduce SD? Otherwise, how can I try?

If CV shows that you have high variability in performance, you should definitely _not_ turn off CV. This indicates that model performance is highly dependent on the splits, which can only be checked for by doing either CV or repetitions.

To reduce the SD and get a more performant model, you should look into splitting strategies that better account for the highly unbalanced nature of the data. Your best option might be to consider using a package like imblanced-learn (https://imbalanced-learn.org/stable/) to generate your own data splits, and then pass them to Chemprop. This isn't something we can hep you setup (our development efforts are focused on v2 at the moment), but we encourage you to open another issue if you run into any roadblocks in using custom splits.

**Comment by JacksonBurns** [comment]
In that case it's possible that the data itself is highly non-smooth, i.e. the structures are not well correlated with the activity or there is a 'cliff' in the molecules included. A more likely candidate is just the dataset size - you might want to consider adding molecular level features to the input to help Chemprop get a 'head start' learning: https://chemprop.readthedocs.io/en/latest/tutorial.html?highlight=generate%20features#additional-features

**Comment by spyda90** [comment]
Hi Jackson, thank you for your comments. I will give it a try!

Francesco
